Define `RegistryEntry` type and use it for dispatch (#31)
giordano/AnalyzeRegistry.jl: docs/src/index.md

# AnalyzeRegistry.jl

The main functionality of the package is the [`analyze`](@ref) function:

```julia
julia> using AnalyzeRegistry

> julia> analyze(find_package("Flux"))

Reviewer: ```suggestion
julia> analyze("Flux")
```
Maybe let's swap the order to show off the convenience?

```

> The argument is a [`RegistryEntry`](@ref), a simple datastructure which points
> to the directory of the package in the registry, where the file `Package.toml`
> is stored.  The function [`find_package`](@ref) gives you the
> [`RegistryEntry`](@ref) of a package in your local copy of any registry, by
> default the [General registry](https://github.com/JuliaRegistries/General).
> 
> *NOTE*: the Git repository of the package will be cloned, in order to inspect
> its content.
> 
> You can also pass the name of a package as argument to `analyze` instead of its
> path: in that case [`find_package`](@ref) will be automatically used to find its
> entry in the registry.

Reviewer: ```suggestion
The argument is a string pointing towards a local path or the name of
a package in a locally-installed registry (the General registry is checked by default).

*NOTE*: the Git repository of the package will be cloned, in order to inspect
its content.

You can also pass a [`RegistryEntry`](@ref), a simple datastructure which points
to the directory of the package in the registry, where the file `Package.toml`
is stored.  The function [`find_package`](@ref) gives you the
[`RegistryEntry`](@ref) of a package in your local copy of any registry, by
default the [General registry](https://github.com/JuliaRegistries/General).
`find_package` is invoked automatically when you pass the name of a package.
```

```julia

> julia> analyze("JuMP")

Reviewer: ```suggestion
julia> analyze(find_package("JuMP"))
```
